### PR TITLE
[CI:BUILD] Packit: install golist before updating downstream spec

### PR DESCRIPTION
**Pull request by lsm5**
The default Packit sandbox environment that runs Packit tasks for
downstream Fedora does not have golist installed by default and can't
run superuser tasks.
    
This commit will download and extract the golist binary from the Fedora
rpm and use it to provide golist.
    
The GOPATH mention in `rpm/update-spec-provides.sh` is only required for
golist to generate the gopaths and doesn't affect upstream or the rpm spec.
    
Currently, the only way to reliably test this is on an open github issue by running
`/packit propose-downstream`. This can't be run on an open PR.
The job-specific packit actions can only be tested via the packit
service and not via packit cli.

@mtrmac @vrothberg PTAL

### CODE REVIEW

File: rpm/update-spec-provides.sh
Comment (mtrmac) [inline rpm/update-spec-provides.sh:18]: I can’t see `golist`, nor the Go standard library, ever referring to `GOPATHDIR`. (Comparing https://pagure.io/go-rpm-macros/blob/master/f/rpm/macros.d/macros.go-rpm#_187 .)

File: rpm/update-spec-provides.sh
Comment (mtrmac) [inline rpm/update-spec-provides.sh:26]: I would expect this to extract to `./usr/bin/golist`, but that’s not in `PATH`, so how does the command get executed?

File: rpm/update-spec-provides.sh
Comment (mtrmac) [inline rpm/update-spec-provides.sh:30]: (BTW looking at what Golist does, it seems that it should be possible to use the standard Go command, something like `go list -deps -f '{{range .Imports}}{{.}}\n{{end}}'  ./...`. I fully support using the same command that `go-rpm-macros` uses, and I know nothing about the environment — quite possibly installing all of Go is even worse than installing `golist` — I just wanted to note the theoretical possibility, now that I took some time to learn what `golist` is.)

**Review by mtrmac — COMMENTED** [review]
Thanks very much for looking at that failure!

Is there a test run somewhere? I’m afraid know nothing about the environment, still, some of the additions are a bit surprising. It’s very likely that’s just my ignorance.

Or is the best way to deal with this to merge and let the next triggered job prove that the script works, because other testing is not all that convenient? That’s fine with me as well :)

File: rpm/update-spec-provides.sh
Comment (lsm5) [reply rpm/update-spec-provides.sh:26]: I need to switch to `./usr/bin/golist`. Thanks for pointing that out. I'll check if I can verify this in the packit local reproducible environment as well.

File: rpm/update-spec-provides.sh
Comment (lsm5) [reply rpm/update-spec-provides.sh:30]: ack. I haven't checked with packit upstream about this, but golang might be a heavy dependency to include in their sandbox. So, I'll keep golist for now.

**Comment by lsm5** [comment]
> Or is the best way to deal with this to merge and let the next triggered job prove that the script works, because other testing is not all that convenient? That’s fine with me as well :)

There's a local test environment, but I haven't yet fully digested it. I'll work on that first to avoid future surprises. Thanks

**Comment by mtrmac** [comment]
I really know nothing about this; if you as the expert think merging & testing live is the most productive approach, I’m totally fine with it. I just wanted to understand what the plan is.

File: rpm/update-spec-provides.sh
Comment (mtrmac) [inline rpm/update-spec-provides.sh:31]: Maybe
```suggestion
GO_IMPORTS=$(GOPATH=$GOPATH ./usr/bin/golist --imported --package-path github.com/containers/$PACKAGE --skip-self | sort -u | xargs "-I{}" echo "Provides: bundled(golang({}))")
```
? `GOPATH` isn’t exported.

**Review by mtrmac — APPROVED** [review]
Thanks!

LGTM, please merge whenever you think is a good time.

**Comment by lsm5** [comment]
@mtrmac Thank you, I'll keep this open for now and merge it once I have a better understanding of the local testing env.

**Comment by lsm5** [comment]
I checked with packit folks, the only way so far to test this change would be to run `slash-packit propose-downstream` on a github issue (can't be done on an open PR). And given I have greenlight from @mtrmac and that it doesn't affect upstream, I'll merge this once tests pass and try running the propose-downstream task on the github issue. I'll also unlink #2036 so it remains open after merging this PR.

**Pull request merged**
